src/settings/FixedDueDateSchedule/FixedDueDateScheduleForm.test.js

```diff
@@ -0,0 +1,512 @@
+      expect(container.getByText(messageIds.saveAndClose)).toBeInTheDocument();
+    });
+
+    it('"Button" component should be called with updated state after clicking on delete button', () => {
+      const deleteButton = container.getByText(messageIds.delete);
+      const expectedProps = {
+        id: 'clickable-delete-item',
+        disabled: !defaultState.confirmDelete,
+      };
+
+      fireEvent.click(deleteButton);
+
+      expect(Button).toHaveBeenNthCalledWith(buttonCallOrder.updatedDelete, expect.objectContaining(expectedProps), {});
+    });
+
+    describe('ConfirmationModal', () => {
+      const modalCallOrder = {
+        render: 1,
+        update: 3,
+      };
+
+      it('"ConfirmationModal" should be called with correct props', () => {
+        const expectedProps = {
+          id: 'deletefixedduedateschedule-confirmation',
+          open: defaultState.confirmDelete,
+          onCancel: expect.any(Function),
+          onConfirm: expect.any(Function),
+        };
+
+        expect(ConfirmationModal).toHaveBeenNthCalledWith(modalCallOrder.render, expect.objectContaining(expectedProps), {});
+      });
+
+      it('"ConfirmationModal" should have correct header', () => {
+        expect(container.getByText(messageIds.deleteHeader)).toBeInTheDocument();
+      });
+
+      it('"ConfirmationModal" should have correct message', () => {
+        expect(container.getByText(messageIds.deleteMessage)).toBeInTheDocument();
+      });
+
+      it('"ConfirmationModal" should have correct label of confirm button', () => {
+        expect(container.getByText(messageIds.deleteLabel)).toBeInTheDocument();
+      });
+
+      it('"onRemove" prop should be triggered after clicking on confirm button', () => {
+        fireEvent.click(container.getByText(messageIds.deleteLabel));
+
+        expect(defaultProps.onRemove).toBeCalledWith(initialValues);
+      });
+
+      it('"ConfirmationModal" should be called with correct props after clicking on "Cancel" button', () => {
+        fireEvent.click(container.getByText(messageIds.delete));
+        fireEvent.click(container.getByText('Cancel'));
+
+        expect(ConfirmationModal).toHaveBeenNthCalledWith(modalCallOrder.update, expect.objectContaining({ open: false }), {});
+      });
+    });
+  });
+
+  describe('Validation check', () => {
+    const initialValues = {
+      id: 'test-id',
+      name: 'test-name',
+    };
+
+    const props = {
+      ...defaultProps,
+      initialValues,
+    };
+
+    beforeEach(() => {
+      render(
+        <FixedDueDateScheduleForm {...props} />
+      );
+    });
+
+    describe('With provided name and without response data', () => {
+      beforeAll(() => {
+        Field.mockImplementationOnce(({ validate }) => {
+          const validateName = async () => {
+            await validate(name);
+          };
+          validateName();
+
+          return null;
+        });
+
+        global.fetch = jest.fn(() => {
```

Comment:
please make sure that we do not need to restore original fetch method after tests. I mean make sure that overriding fetch method does not affect tests in other test files. Maybe we need to restore original fetch method. Please read about this.

Reply:
It is not necessary. We have separated `fetch` mocking for each `describe` and each `describe` has only one test.